src/Adapters/JobsAdapter.ts

```diff
@@ -0,0 +1,156 @@
+        return this.adtHostUrl;
+    }
+
+    createJob = async (params: AdapterMethodParamForCreateJobs) => {
+        const adapterMethodSandbox = new AdapterMethodSandbox(this.authService);
+
+        return adapterMethodSandbox.safelyFetchDataCancellableAxiosPromise(
+            ADTJobsData,
+            {
+                method: 'put',
+                url: this.generateUrl(
+                    `/jobs/${encodeURIComponent(params.jobId)}`
+                ),
+                headers: this.generateHeaders(),
+                params: {
+                    'api-version': ADT_ApiVersion,
+                    inputbloburi: params.inputBlobUri,
+                    outputbloburi: params.outputBlobUri
+                }
+            }
+        );
+    };
+
+    deleteJob = async (params: AdapterMethodParamsForJobs) => {
+        const adapterMethodSandbox = new AdapterMethodSandbox(this.authService);
+
+        return adapterMethodSandbox.safelyFetchDataCancellableAxiosPromise(
+            ADTJobsData,
+            {
+                method: 'delete',
+                url: this.generateUrl(
+                    `/jobs/${encodeURIComponent(params.jobId)}`
+                ),
+                headers: this.generateHeaders(),
+                params: {
+                    'api-version': ADT_ApiVersion
+                }
+            }
+        );
+    };
+
+    cancelJob = async (params: AdapterMethodParamsForJobs) => {
+        const adapterMethodSandbox = new AdapterMethodSandbox(this.authService);
+
+        return adapterMethodSandbox.safelyFetchDataCancellableAxiosPromise(
+            ADTJobsData,
+            {
+                method: 'post',
+                url: this.generateUrl(
+                    `/jobs/${encodeURIComponent(params.jobId)}`
+                ),
+                headers: this.generateHeaders(),
+                params: {
+                    'api-version': ADT_ApiVersion
+                }
+            }
+        );
+    };
+
+    getAllJobs = async () => {
+        const adapterMethodSandbox = new AdapterMethodSandbox(this.authService);
+
+        return adapterMethodSandbox.safelyFetchDataCancellableAxiosPromise(
+            ADTJobsData,
+            {
+                method: 'get',
+                url: this.generateUrl(`/jobs/imports`),
+                headers: this.generateHeaders(),
+                params: {
+                    'api-version': ADT_ApiVersion
+                }
+            }
+        );
+    };
+}
+export default interface JobsAdapter extends ADTAdapter, ADT3DSceneAdapter {
+    getAllJobs: () => Promise<AdapterResult<any>>;
+    cancelJob: (
+        params: AdapterMethodParamsForJobs
+    ) => Promise<AdapterResult<any>>;
+    deleteJob: (
+        params: AdapterMethodParamsForJobs
+    ) => Promise<AdapterResult<any>>;
+    createJob: (
+        params: AdapterMethodParamForCreateJobs
+    ) => Promise<AdapterResult<any>>;
+}
+applyMixins(JobsAdapter, [ADT3DSceneAdapter]);
```

Comment:
why dont you add ADTAdapter to this mixin array so that you will not need to implement ADTAdapter methods like getAdtHostUrl or setAdtHostUrl

Reply:
okay, second eye. Since ADT3DSceneAdapter is already extending ADTAdapter, you should not need to extend it again.

Reply:
Also, make sure that you want to extend ADT3DSceneAdapter, not ADTAdapter. If you only need ADTAdapter capabilities, you should only extend from ADTAdapter and apply to mixin to ADTAdapter only.